.vib/spring-cloud-dataflow/cypress/cypress/integration/spring-cloud-dataflow_spec.js

```diff
@@ -0,0 +1,107 @@
+/// <reference types="cypress" />
+import { random } from './utils.js';
+import {
+  importAStreamApplication,
+  importATaskApplication,
+  createATask,
+} from './prepare-app-state.js';
+
+it('allows getting Spring Cloud Dataflow info', () => {
+  cy.visit('/dashboard');
+  cy.get('.signpost-trigger').click();
+  cy.contains('.signpost-content-body', 'Version');
+  cy.contains('button', 'More info').click();
+  cy.contains('.modal-content', 'Core')
+    .and('contain', 'Dashboard')
+    .and('contain', 'Shell');
+});
+
+it('allows a stream to be created and deployed', () => {
+  const STREAM = 'mongodb | cassandra';
+
+  importAStreamApplication();
+  cy.visit('/dashboard');
+  cy.get('[routerlink="streams/list"').click();
+  cy.contains('.btn-primary', 'Create stream(s)').click();
+  cy.get('.CodeMirror-line').type(STREAM);
+  cy.contains('#v-2', 'mongodb').and('contain', 'cassandra');
+  cy.contains('button', 'Create stream(s)').click();
+  cy.get('.modal-content').should('contain', 'Create Stream');
+  cy.fixture('streams').then((stream) => {
+    cy.get('input[placeholder="Stream Name"]').type(
+      `${stream.newStream.name}-${random}`
+    );
+    console.log(random);
+    cy.contains('.btn-primary', 'Create the stream').click();
+    cy.contains(
+      '.datagrid-inner-wrapper',
+      `${stream.newStream.name}-${random}`
+    );
+  });
+  cy.contains('.toast-container', 'successfully');
+  cy.contains('clr-dg-cell', 'UNDEPLOYED')
+    .siblings('clr-dg-cell', 'test-stream')
+    .first()
```

Comment:
Hmm relying on the order of an array wrt to an element in the UI does not sound very resilient. Is there any other possibility to avoid this?

Sometimes, when is difficult to come up with the right global selector, it is useful to first match the div/table containing our desired element and then use the cypress function `within`.

Reply:
It's not really relevant which element we click. 
It is a table and I want to take any element. It can be the first or the last since they are all the same. 
I am taking the last of the elements from the table which has the tag `UNDEPLOYED`.  It's just a convenient use of the `siblings()` method. 
I can show you over a call and we can decide together if it's appropriate or not.